### ISSUE TITLE
TypeScript types not recognized

### ISSUE THREAD

**Issue by bengry** [graphql-mocks/graphql-mocks#259]
TypeScript types are not recognized in the latest version (`v0.11.0`). I don't think it was the case before, but I might be wrong.

To fix this I applied this patch (locally using `yarn patch`):
```diff
diff --git a/package.json b/package.json
index 3bca561d35e4191b2c403aed38995e3d3957f3ee..eab379e388b8d481d7978071424ae2569cab2f0d 100644
--- a/package.json
+++ b/package.json
@@ -9,59 +9,73 @@
   "exports": {
     ".": {
       "import": "./es/index.mjs",
-      "require": "./index.js"
+      "require": "./index.js",
+      "types": "./index.d.ts"
     },
     "./graphql": {
       "import": "./es/graphql/index.mjs",
-      "require": "./graphql/index.js"
+      "require": "./graphql/index.js",
+      "types": "./graphql/index.d.ts"
     },
     "./graphql/type-utils": {
       "import": "./es/graphql/type-utils/index.mjs",
-      "require": "./graphql/type-utils/index.js"
+      "require": "./graphql/type-utils/index.js",
+      "types": "./graphql/type-utils/index.d.ts"
     },
     "./graphql/utils": {
       "import": "./es/graphql/utils/index.mjs",
-      "require": "./graphql/utils/index.js"
+      "require": "./graphql/utils/index.js",
+      "types": "./graphql/utils/index.d.ts"
     },
     "./highlight": {
       "import": "./es/highlight/index.mjs",
-      "require": "./highlight/index.js"
+      "require": "./highlight/index.js",
+      "types": "./highlight/index.d.ts"
     },
     "./highlight/utils": {
       "import": "./es/highlight/utils/index.mjs",
-      "require": "./highlight/utils/index.js"
+      "require": "./highlight/utils/index.js",
+      "types": "./highlight/utils/index.d.ts"
     },
     "./resolver-map": {
       "import": "./es/resolver-map/index.mjs",
-      "require": "./resolver-map/index.js"
+      "require": "./resolver-map/index.js",
+      "types": "./resolver-map/index.d.ts"
     },
     "./resolver-map/utils": {
       "import": "./es/resolver-map/utils/index.mjs",
-      "require": "./resolver-map/utils/index.js"
+      "require": "./resolver-map/utils/index.js",
+      "types": "./resolver-map/utils/index.d.ts"
     },
     "./resolver": {
       "import": "./es/resolver/index.mjs",
-      "require": "./resolver/index.js"
+      "require": "./resolver/index.js",
+      "types": "./resolver/index.d.ts"
     },
     "./resolver/utils": {
       "import": "./es/resolver/utils/index.mjs",
-      "require": "./resolver/utils/index.js"
+      "require": "./resolver/utils/index.js",
+      "types": "./resolver/utils/index.d.ts"
     },
     "./relay": {
       "import": "./es/relay/index.mjs",
-      "require": "./relay/index.js"
+      "require": "./relay/index.js",
+      "types": "./relay/index.d.ts"
     },
     "./pack": {
       "import": "./es/pack/index.mjs",
-      "require": "./pack/index.js"
+      "require": "./pack/index.js",
+      "types": "./pack/index.d.ts"
     },
     "./pack/utils": {
       "import": "./es/pack/utils/index.mjs",
-      "require": "./pack/utils/index.js"
+      "require": "./pack/utils/index.js",
+      "types": "./pack/utils/index.d.ts"
     },
     "./wrapper": {
       "import": "./es/wrapper/index.mjs",
-      "require": "./wrapper/index.js"
+      "require": "./wrapper/index.js",
+      "types": "./wrapper/index.d.ts"
     }
   },
   "license": "MIT",
```

It would be great if this could be fixed in the published package.

**Comment by chadian** [comment]
Thanks for reporting, I was able to reproduce this. Opened #260 to get this work started and should have it wrapped up and merged tomorrow with new packages published.

**Comment by chadian** [comment]
@bengry since you're actively using the project I'd love to hear if you have any general feedback or feel like it's missing anything useful. Cheers!

**Comment by bengry** [comment]
> @bengry since you're actively using the project I'd love to hear if you have any general feedback or feel like it's missing anything useful. Cheers!

We recently started using it for our integration tests. We heavily rely on GraphQL in our portal, and `graphql-mocks` helps us achieve two things:
1. Keep test authoring easy - we use a (custom) middleware that uses [faker.js](https://github.com/faker-js/faker) to generate consistent mocks, based on a custom seed, so our data is consistent across runs, yet not as simplistic as [`@graphql-tools/mock`](https://the-guild.dev/graphql/tools/docs/mocking)'s naive approach.
2. Allow us to do visual regression testing on the portal, which is a direct result of the consistent data mentioned above.

I think the main things that are a bit annoying are:
1. Using `graphql-mocks` seems to slow the startup time of tests. We use Playwright and from the time I start running a test until the Chromium browser is opened is quite substantial. I think I mentioned this in another reply in this repo in the past, but I haven't gotten around to perf. testing exactly what's the cause. It might just be that we load & parse our entire schema (~45K LOC for the `.graphql` schema file) in a blocking way, but I'm not 100% sure that's the cause.
2. When used naively, the first GQL query that hits the middleware is much slower, since [a lot of work](https://github.com/graphql-mocks/graphql-mocks/blob/35f94746b026449134d643ff89f84d9bd0c136c6/packages/falso/src/falso-middleware.ts#L15-L42) is done in that first call. In an attempt to alleviate that issue, I've moved some of this logic to be done ahead of the first GQL call, but this required me to pass in the `GraphQLSchema` twice - once in `graphqlSchema` and once to my middleware.
[Here's the code](https://gist.github.com/bengry/2e76d76a29da12ab8f5907c4dfc933c8) I'm using to achieve our consistent mocking. I've had to replace `/` with `\` there to make GitHub happy with the file names, but I hope it gives an idea of what we have.

P.S.
Even though `@graphql-tools/mock` is the de-facto standard solution for GraphQL mocks, it seems like it's not possible to use custom logic that's based on the query itself, which is a shame (I tried asking about it [here](https://github.com/ardatan/graphql-tools/discussions/6317), but got no response).

**Comment by chadian** [comment]
Thanks for the insights. I've created #273 to continue the conversation because I have some ideas that I think would make some improvements.

For this issue, across all packages I've published updated `"types"` in each package.json. I tested in `.cjs` and `.mjs` files and was able to reproduce the issue and the fix. Let me know if this is still causing you an issue.

**Comment by bengry** [comment]
Fixed in v0.11.1
